pyfstat/grid_based_searches.py

```diff
@@ -157,11 +156,11 @@ def inititate_search_object(self):
                 detectors=self.detectors,
                 injectSources=self.injectSources,
             )
-
-            def cut_out_tstart_tend(*vals):
-                return self.search.get_semicoherent_det_stat(*vals[2:])
-
-            self.search.get_det_stat = cut_out_tstart_tend
+            self.search.get_det_stat = self.search.get_semicoherent_det_stat
+        # make sure to overwrite the min/max starttime in case the user
+        # passed None and they were read from SFTs
+        self.minStartTime = self.search.minStartTime
+        self.maxStartTime = self.search.maxStartTime
```

Comment:
This is what should solve #167 and was impossible to do without the bigger changes or a similarly huge reordering.